### PR TITLE
Updated DoctrineModule authentication doc with some ZF3-related information and examples

### PR DESCRIPTION
**Pull request by vvaswani**

### CODE REVIEW

File: docs/authentication.md [hunk 1]
```diff
@@ -6,7 +6,7 @@ Authentication through Doctrine is fully supported by DoctrineModule through an
 
 In order to authenticate a user (or anything else) against Doctrine, the following workflow will be use :
```
Comment (TomHAnderson) [inline docs/authentication.md]: `will be use :` to `is used:`

File: docs/authentication.md [hunk 2]
```diff
@@ -95,11 +123,30 @@ class Module
...
+namespace Application\Factory\Controller;
+
+use Interop\Container\ContainerInterface;
+use Zend\ServiceManager\Factory\FactoryInterface;
+
+class ApplicationControllerFactory implements FactoryInterface
+{
+    public function __invoke(ContainerInterface $container, $requestedName, array $options = null)
+    {
+        $as = $container->get('doctrine.authenticationservice.orm_default');
```
Comment (TomHAnderson) [inline docs/authentication.md]:
```
$authenticationService = $container->get('doctrine.authenticationservice.orm_default');

return new $requestedName($authenticationService);
```

Brevity does not read like english; the abbreviated term is defined differently.

File: docs/authentication.md [hunk 3]
```diff
@@ -95,11 +123,30 @@ class Module
...
+        return new $requestedName($as);
+    }
+}
+```
 
 #### Using the AuthenticationService
 
-Now that we have defined how to create a `Zend\Authentication\AuthenticationService` object, we can use it in our code. For more information about Zend authentication mechanisms, please read [the ZF 2 Authentication's documentation](http://framework.zend.com/manual/2.0/en/modules/zend.authentication.intro.html).
+Now that we have defined how to create a `Zend\Authentication\AuthenticationService` object, we can use it in our code. For more information about Zend authentication mechanisms, please read [the ZF 2 Authentication's documentation](http://framework.zend.com/manual/2.4/en/modules/zend.authentication.intro.html).
```
Comment (TomHAnderson) [inline docs/authentication.md]: Too many commas.  Remove `object,` and `, please read` commas.

**Comment by TomHAnderson** [comment]
It is good practice to create a branch on your clone instead of pulling from your own master.

**Comment by vvaswani** [comment]
@TomHAnderson, changes made. Thanks for the note about using a clone branch, I'm new to PRs so not fully aware of how things work.

**Pull request merged**